### PR TITLE
fix: Explicitly check for browser env

### LINKED ISSUE #173: Support Vercel Edge environment
Hey folks, I'm developing an app using NextJS and I have a function that runs on the edge, and when I try to use the fauna client in this function, I get the following error:
```
Critical dependency: require function is used in a way in which dependencies cannot be statically extracted

Import trace for requested module:
./node_modules/fauna/dist/browser/index.js
Error [ReferenceError]: navigator is not defined
```
I guess it is because it tries to access some browser feature that is not available 🤔 

To reproduce this you create a NextJS project and inside of the app folder, create a `/api/route.ts` with something like:
```ts
export const runtime = "edge";

export function GET() {
    const fauna = new Client({ ... })
    fauna.query(fql`[]`)
}
```
Sorry for not having too much more info to help with debug 😞

### PR DESCRIPTION
**Pull request by ptpaterson**
Ticket(s): [FE-3729](https://faunadb.atlassian.net/browse/FE-3729)

Fixes #173 

Leaving this as a draft while we get some testing done.

## Problem
We currently assume that if you are not running in Node or Service Worker, you are in a browser. That blows up in environments like Vercel's Edge Runtime and Fastly's Compute@Edge.

## Solution
Explicitly check for browser env before getting details from `navigator` object. If none of our env checks are true, then report unknown.

## Testing
Running locally with some edge projects.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

[FE-3729]: https://faunadb.atlassian.net/browse/FE-3729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by cleve-fauna** [comment]
seems OK without hardcore testing.

File: src/util/environment.ts [hunk 1]
```diff
@@ -19,25 +19,51 @@ export const getDriverEnv = (): string => {
     runtime: "unknown",
   };
 
-  let isServiceWorker = false;
-  try {
-    isServiceWorker = global instanceof ServiceWorkerGlobalScope;
-  } catch (_) {
-    // okay if this fails
-  }
```
Comment (cleve-fauna) [inline src/util/environment.ts:27]: are we losing some safety?

**Review by cleve-fauna — COMMENTED** [review]
Wrapping this whole process in some kind of try catch so we don't blow up on writing this data to headers would be nice if we don't have it already.

File: src/util/environment.ts (hunk 1 above)
Comment (ptpaterson) [reply src/util/environment.ts:27]: No, we should be fine. I found recommendations for checking for Service Worker env that don't rely on trying something that can throw. This try/catch block was only around using `instanceof`. `typeof` doesn't throw an error if the variable isn't declared.

**Review by cleve-fauna — DISMISSED** [review]

**Review by cleve-fauna — COMMENTED** [review]
Discussed on slack - this telemetry could still cause the driver to be unusuable in idiosyncratic envs. Plan is to wrap in try catch to prevent that.

**Review by cleve-fauna — APPROVED** [review]

**Pull request merged**